Lib/unittest/test/testmock/testwaitablemock.py

```diff
@@ -0,0 +1,144 @@
+import threading
+import time
+import unittest
+
+from test.support import start_threads
+from unittest.mock import patch, WaitableMock, call
+
+
+class Something:
+
+    def method_1(self):
+        pass
+
+    def method_2(self):
+        pass
+
+
+class TestWaitableMock(unittest.TestCase):
+
+
+    def _call_after_delay(self, func, *args, delay):
+        time.sleep(delay)
```

Comment:
This might be painful but it might be better to use events to wait for things to happen rather than relying in sleep. Otherwise, this might be adding considerable time to the test suit.

I'd say wait for a core dev to comment about it, it might be fine though.

Reply:
I have kept the delays 0.1-0.5s to simulate latency of the call. Currently, it runs on 2.5 seconds on my local machine and might increase once I add more tests. I am new to using event objects so if there is a better way to simulate latency then I can refactor my tests.